src/model/Document.cpp

```diff
@@ -537,6 +534,7 @@ XojPdfDocument& Document::getPdfDocument()
 	return this->pdfDocument;
 }
 
+//Todo: the locking is broken, fix it
```

Comment:
details?

Reply:
The problem is, that the lock and the unlock is not in the same scope. And this sort of synchronization handling is bug prone and unsafe to exceptions or code changes. So the hole part should be refactored some time.